crates/ruff_python_formatter/src/other/arguments.rs

```diff
@@ -170,7 +190,9 @@ impl FormatNodeRule<Arguments> for FormatArguments {
             + kwonlyargs.len()
             + usize::from(kwarg.is_some());
 
-        if num_arguments == 0 {
+        if self.parentheses == ArgumentsParentheses::SkipInsideLambda {
+            group(&format_inner).fmt(f)?;
```

Comment:
Can we add an assertion that `dangling_comments` is empty (is this guaranteed?)

Reply:
e11be20805a7e6cbd2492b7af2483f317202677b

Reply:
> is this guaranteed?

Let me think about this later today.

Reply:
My initial thought is it should be since afaik `lambda (` is a syntax error.

```
SyntaxError: Unexpected token '('Ruff(E999)
```

But a couple more thoughts:
- I may not be thinking about *assignment* of dangling comments for `lambda`s.
- Maybe I can run this on more code with an assertion.

Reply:
I managed to construct this example:
```python
a = (
    lambda  # a
           : 1
)
```
This is a pronouncedly strange edge case (it does neither make sense to have a comment there nor to break the empty lambda argument list over two lines), i.e. it doesn't matter where the comment ends up as long as it's stable, but currently we lose it which we shouldn't:
```
{
    Node {
        kind: Arguments,
        range: 10..36,
        source: `lambda  # a⏎`,
    }: {
        "leading": [],
        "dangling": [
            SourceComment {
                text: "# a",
                position: EndOfLine,
                formatted: false,
            },
        ],
        "trailing": [],
    },
}
a = lambda: 1
```

Reply:
Perfect. Thank you. I'll try to handle this today.

Reply:
dcb5cea49bb87514d77fe3faae217bf1095b0ce9